tests/integration/api_swarm_test.py

```diff
@@ -111,7 +116,8 @@ def test_update_swarm(self):
         assert self.init_swarm()
         swarm_info_1 = self.client.inspect_swarm()
         spec = self.client.create_swarm_spec(
-            snapshot_interval=5000, log_entries_for_slow_followers=1200,
+            name='default', snapshot_interval=5000,
+            log_entries_for_slow_followers=1200,
```

Comment:
Should be we doing an init_swarm() here before calling update() ? @thaJeztah

Reply:
`init_swarm` is called here: https://github.com/docker/docker-py/pull/1829/files#diff-5045b9ca2e2c5acfebac3a4ce041b23aR116

Reply:
Right, I guess I meant doing init_swarm() with a name specified. But I guess it wouldn't matter. Keeping this change and reverting others.

Reply:
Can we make the change in moby/moby so that the default is set there if omitted? It would still produce an error if any name _other than_ "default" is given, but at least it won't be a breaking change for users of the API.

TL;DR no changes in the docker-py tests should be needed, and if there are, it means we introduced a breaking change

Reply:
I'm OK with setting the default name in create_swarm_spec().

The correct semantics for updating swarmkit objects is read-modify-write. Clients should not be writing cluster spec without reading it first i.e. the spec on the client side should be exactly the same as the spec in swarmkit.

Reply:
I'm okay with _adding_ a test to test that updating both _with_ and _without_ a name set works, but want to make sure that the current behaviour doesn't break (if `name` was intended to be a required parameter, we should've enforced that long ago; unfortunately we didn't, so we'll have to deal with that)

Reply:
Name is *not* a required param. What is required is that you *don't change it*, basically always read it before writing.

Read-modify-write is not really a new thing and I believe has always been the semantic. If clients always do that, then this should not be breaking. 

The problem in the failing test is that it tries to update the spec without reading it first.